server/src/main/java/com/cloud/deploy/DeploymentPlanningManagerImpl.java

```diff
@@ -548,6 +547,78 @@ public DeployDestination planDeployment(VirtualMachineProfile vmProfile, Deploym
         return dest;
     }
 
+    /**
+     * Adds disabled resources (Data centers, Pods, Clusters, and hosts) to exclude list (avoid) in case of disabled state.
+     */
+    public void avoidDisabledResources(VirtualMachineProfile vmProfile, DataCenter dc, ExcludeList avoids) {
+        if (vmProfile.getType().isUsedBySystem() && isSystemVmDeployableInDisabledResources()) {
+            return;
+        }
+
+        VMInstanceVO vm = _vmInstanceDao.findById(vmProfile.getId());
+        AccountVO owner = accountDao.findById(vm.getAccountId());
+        if (owner.getRoleId() == ADMIN_ACCOUNT_ROLE_ID && isAdminVmDeployableInDisabledResources()) {
+            return;
+        }
+
+        avoidDisabledDataCenters(dc, avoids);
+        avoidDisabledPods(dc, avoids);
+        avoidDisabledClusters(dc, avoids);
+        avoidDisabledHosts(dc, avoids);
+    }
+
+    /**
+     * Returns the value of the ConfigKey 'allow.system.vm.disabled.resource'. This method allows mocking and testing with the respective ConfigKey parameter.
```

Comment:
"Returns the value of the ConfigKey 'allow.system.vm.ON.disabled.resource" - since name was changed.